core_backend/app/routers/question_answer.py

```diff
@@ -43,55 +44,63 @@ async def llm_response(
     LLM response creates a custom response to the question using LLM chat and the
     most similar embeddings to the user query in the vector db.
     """
+    (
+        user_query_db,
+        user_query_refined,
+        response,
+    ) = await get_user_query_and_response(user_query, asession)
 
-    feedback_secret_key = generate_secret_key()
+    response = await get_llm_answer(user_query_refined, response, qdrant_client)
+    await save_query_response_to_db(asession, user_query_db, response)
+
+    return response
+
+
+@classify_safety
```

Comment:
We probably don't need all these but adding these here so we can test it out if we want to.